write-fonts/src/tables/glyf.rs

```diff
@@ -0,0 +1,298 @@
+    fn get(self) -> (i16, i16) {
+        self
+    }
+}
+
+// adapted from simon:
+// https://github.com/simoncozens/rust-font-tools/blob/105436d3a617ddbebd25f790b041ff506bd90d44/otmath/src/lib.rs#L17
+fn ot_round(val: f32) -> i16 {
+    (val + 0.5).floor() as i16
+}
+
+impl Contour {
+    /// Create a new contour begining at the provided point
+    pub fn new(pt: impl OtPoint) -> Self {
+        let (x, y) = pt.get();
+        Self(vec![CurvePoint::on_curve(x, y)])
+    }
+
+    /// The total number of points in this contour
+    pub fn len(&self) -> usize {
+        self.0.len()
+    }
+
+    /// `true` if this contour is empty
+    pub fn is_empty(&self) -> bool {
+        self.0.is_empty()
+    }
+
+    /// Add a line segment
+    pub fn line_to(&mut self, pt: impl OtPoint) {
+        let (x, y) = pt.get();
+        self.0.push(CurvePoint::on_curve(x, y));
+    }
+
+    /// Add a quadratic curve segment
+    pub fn curve_to(&mut self, p0: impl OtPoint, p1: impl OtPoint) {
+        let (x0, y0) = p0.get();
+        let (x1, y1) = p1.get();
+        self.0.push(CurvePoint::off_curve(x0, y0));
+        self.0.push(CurvePoint::on_curve(x1, y1));
+    }
+}
+
+impl SimpleGlyf {
+    /// Attempt to create a simple glyph from a kurbo `BezPath`
+    ///
+    /// The path may contain only line and quadratic bezier segments. The caller
+    /// is responsible for converting any cubic segments to quadratics before
+    /// calling.
+    ///
+    /// Returns an error if the input path is malformed; that is, if it is empty,
+    /// contains cubic segments, or does not begin with a 'move' instruction.
+    //TODO: figure out a more general API? maybe a nested builder thing, where you
+    //build contours, and from those contours build a glyph? idk?
+    pub fn from_kurbo(path: &BezPath) -> Result<Self, BadKurbo> {
+        let mut contours = Vec::new();
+        let mut current = None;
+
+        for el in path.elements() {
+            match el {
+                kurbo::PathEl::MoveTo(pt) => {
+                    if let Some(prev) = current.take() {
+                        contours.push(prev);
+                    }
+                    current = Some(Contour::new(*pt));
+                }
+                kurbo::PathEl::LineTo(pt) => {
+                    current.as_mut().ok_or(BadKurbo::MissingMove)?.line_to(*pt)
+                }
+                kurbo::PathEl::QuadTo(p0, p1) => current
+                    .as_mut()
+                    .ok_or(BadKurbo::MissingMove)?
+                    .curve_to(*p0, *p1),
+                kurbo::PathEl::CurveTo(_, _, _) => return Err(BadKurbo::HasCubic),
+                // I think we can just ignore this, and remove duplicate points
+                // at the end?
+                kurbo::PathEl::ClosePath => (),
+            }
+        }
+
+        contours.extend(current);
+
+        for contour in &mut contours {
+            //TODO: verify that single-point contours are actually meaningless?
```

Comment:
I recall something about fonts with `sbix` tables using single point contours to represent image offsets. @behdad might have more info on this.

Reply:
They use two points, to represent the glyph bounding box.

Reply:
Being meaningless and being an error condition aren't quite the same thing. Are we sure this should be an *error*?